identity-account/src/identity/chain_state.rs

```diff
@@ -0,0 +1,63 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use serde::Serialize;
+
+use identity_iota::tangle::MessageId;
+use identity_iota::tangle::MessageIdExt;
+
+/// Holds the last published message ids of the integration and diff chains.
+#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
+pub struct ChainState {
+  #[serde(default = "MessageId::null", skip_serializing_if = "MessageId::is_null")]
+  previous_integration_message_id: MessageId,
```

Comment:
I wonder if the `ChainState` could be removed if e.g. `ResolutionMetadata` contains these `message_ids` and is stored with the old state. That would assume the `Account` resolves the document from the Tangle after every update though...

No changes requested, just wondering about the metadata refactor.

Reply:
That would be ideal! I think that should actually work because the last published document (= latest document on tangle) is always stored in storage. Then the two cases should work:

1. In a non-distributed setup, only one device/machine/program updates a given identity, and so taking the message ids from the last published document's metadata would work.
2. In a distributed setup, a valid update is only possible if the latest state has been synchronized before, which is why we require developers of such setups to call sync before publishing an update. Then we'd also have the latest message ids locally. If someone fails to synchronize before, then no matter if we hold the state in `ChainState`, or if we take it from the metadata, it will be outdated and the update will be ignored.

Reply:
That makes sense. I'm just thinking about other fields that depend on resolution (like `version`) that cannot/should not be updated locally. Anyway, I'll keep this in mind during implementation.